include/task_manager.h

```diff
@@ -164,10 +109,10 @@ namespace detail {
 
 		task_manager::delegate* m_delegate;
 
-		const size_t m_num_collective_nodes;
+		size_t m_num_collective_nodes;
 		policy_set m_policy;
 
-		task_graph m_tdag;
+		task_graph* m_tdag;
```

Comment:
```suggestion
		task_graph& m_tdag;
```

Should this be a reference? We don't need to copy `task_manager` and this can never be null.

Reply:
The lint goes away when I delete all copy ctors and assignment operators, so I did that with a comment to eventually PIMPL this class.